doc.txt

```diff
@@ -31,3 +31,13 @@
 /**
  * @}
  */
+
+/**
+ * @defgroup    examples Examples
+ * @brief       Example applications
+ */
+
+/**
+ * @defgroup    tests Tests
+ * @brief       Tests applications
```

Comment:
AFAIK those groups will be empty in the rendered doc. Do we really need them?

Reply:
Yes, all examples are using `@ingroup examples` and tests `@ingroup tests`. I know this is not nice but at least it silents the check script.
There are 3 solutions:
1. Remove all `@ingroup tests` and `@ingroup examples`
2. Ignore undefined examples and tests groups in the check script
3. Define `tests` and `examples` groups and document them. If this solution is pushed a bit, we could have all examples and tests documented in doxygen (instead of READMEs).

I think `1.` should be avoided because it will touch too many files and maybe one day it can be useful to have those groups defined.
`2.` is the simplest.
`3.` is IMHO technically and from a centralized documentation the best solution. But requires more work.

Reply:
I think 3 does make little sense. While I prefer API documentation and usage documentation to be centralized, the READMEs of applications should be kept *with the application* for two reasons:

1. One can easily open a README in command-line or their favorite editor/IDE while working with the example and does not need to open an extra browser window
2. The context might get lost if the README is moved somewhere else

As such, I prefer 2, but instead of ignoring the *groups* I rather would ignore the *directories* (as e.g. the vendor headers).

Furthermore, I don't see the point of being so strict with group names in .c-files, as they are not parsed by doxygen, so they can't break the rendered doc (which was the whole point of my initial issue), so it might also be sensible to ignore .c-files altogether. But maybe I'm in the minority with this opinion.

Reply:
I dropped the defined examples and tests groups. Will update #9266 with solution 2.